src/requirements/WalletActivity/components/ControlledNumberInput.tsx

```diff
@@ -34,11 +50,96 @@ const ControlledNumberInput = ({
     },
   })
 
+  const decimalsLimit = decimalsMax < MAX_DECIMALS ? decimalsMax : MAX_DECIMALS
+
+  /**
+   * Makes the step size adapt to the precision. E.g., if the entered number is
+   * 1.001, the step size will be set to 0.001, therefore the next step up will be
+   * 1.002.
+   */
+  const updateStepSize = useCallback(
```

Comment:
I think we could just use a `useMemo` here, so we wouldn't need a `useState`, a function wrapped in `useCallback` & a `useEffect` just to handle the `stepSize`